watchdog_reader.go

```diff
@@ -23,9 +25,14 @@ func newWatchdogReader(reader io.Reader, timeout time.Duration, timer *time.Time
 
 // Read reads up to len(p) bytes into p
 func (t *watchdogReader) Read(p []byte) (n int, err error) {
-	// FIXME limit the amount of data read in one chunk so as to not exceed the timeout?
+	//never read more bytes than watchdogChunkSize
+	readTarget := p
+	if len(p) > watchdogChunkSize {
+		readTarget = p[0:watchdogChunkSize]
+	}
+
```

Comment:
I'm not sure how practical these concerns are. I already chose 1 MB to be comfortably large for most usages (though I guess that we can also up this to 8 MB or so). If you're downloading files that are much larger than that, my guesstimate would be that more time is spent handling the TCP packets than switching contexts.

In general, I don't think it will hurt to keep the chunk size private at first. One can always expose it later if there turns out to be an actual need for that. It's always easier to increase API surface than decrease it. Also, users can just put their own buffering reader in front of the watchdog reader to avoid multiple context switches.